instance/instance.go

```diff
@@ -318,19 +357,35 @@ func unmarshalInstance(reader io.Reader, post bool) (*models.Instance, error) {
 		return nil, errors.New("Failed to parse json body: " + err.Error())
 	}
 
+	if instance.State != "" {
+		if err := models.ValidateInstanceState(instance.State); err != nil {
+			return nil, err
+		}
+	}
+
 	if post {
-		if instance.Links.Job != nil {
-			if instance.Links.Job.ID == "" || instance.Links.Job.HRef == "" {
-				return nil, errors.New("Missing job properties")
-			}
-		} else {
+		log.Debug("got here", log.Data{"instance": instance})
+		if instance.Links == nil {
+			return nil, errors.New("Missing job properties")
+		}
+		if instance.Links.Job == nil {
```

Comment:
Consider combining these two if statements if the outcome is the same error message. 
```
if instance.Links == nil && instance.Links.Job == nil {
   return nil, errors.New("Missing job properties")
}
```